Mathlib/CategoryTheory/Closed/Enrichment.lean

```diff
@@ -0,0 +1,43 @@
+/-
+Copyright (c) 2024 Daniel Carranza. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Carranza
+-/
+import Mathlib.CategoryTheory.Enriched.Basic
+import Mathlib.CategoryTheory.Closed.Monoidal
+
+/-!
+# A closed monoidal category is enriched in itself
+
+From the data of a closed monoidal category C, we define a C-category structure for C
+where the hom-object is given by the internal hom (coming from the closed structure).
+
+The instance is given at the end of the file.
+
+We use `scoped instance` to avoid potential issues where C may also have
+a C-category structure coming from another source (e.g. the type of simplicial sets
+`SSet.{v}` has an instance of `EnrichedCategory SSet.{v}` as a category of simplicial objects;
+see `AlgebraicTopology/SimplicialCategory/SimplicialObject`).
```

Comment:
```suggestion
We use `scoped instance` to avoid potential issues where `C` may also have
a `C`-category structure coming from another source (e.g. the type of simplicial sets
`SSet.{v}` has an instance of `EnrichedCategory SSet.{v}` as a category of simplicial objects;
see `AlgebraicTopology/SimplicialCategory/SimplicialObject`).
```